### PR TITLE
dist: add github label check to drone build script

### PR DESCRIPTION
**Pull request by phiros**
This integrates the github label check into the drone build script.

### CODE REVIEW

**Comment by gebart** [comment]
Could you add a check for an environment variable from outside of the script as well for disabling the check? That way we can override the label behaviour from the Drone control panel if we want to.

**Comment by phiros** [comment]
@gebart done (variable is called ```FORCE_FULL_CHECK```)
I also fixed a small error (all tests should be executed if commits were merged with master aka ```CI_PULL_REQUEST=false``` )

File: dist/tools/drone-scripts/build_and_test.sh [hunk 1]
```diff
@@ -37,12 +37,34 @@ export -f exec_build_func
...
-# Execute all groups in parallel (-k ensures correct ordering of the
-# output)
-parallel -k exec_build_func {} "$@"  ::: static-tests avr8 msp430 x86 arm7 \
-                                         cortex_m0 cortex_m3_1 cortex_m3_2 \
-                                         cortex_m4 \
-|& tee -a "$MYTMPDIR/output.log"
+# Check for PRs and labels
+FULL_CHECK=true
+if [ -z "$FORCE_FULL_CHECK" ]; then
+    if [ $TRAVIS_PULL_REQUEST != false ] && ! [ -z "$TRAVIS_PULL_REQUEST" ]; then
```
Comment (gebart) [inline dist/tools/drone-scripts/build_and_test.sh]: `s/TRAVIS_/CI_/`
Gives the most "correct" variable names in this context, right?

**Comment by gebart** [comment]
Changes look good, though I still haven't had time to get my drone install up and running again, untested ACK when the above comment is addressed.
OK to squash.

**Comment by phiros** [comment]
Ok squashed.

BTW: I am already using this script for another CI system (strider CD).
So far it works ok-ish (about 60 min. for a complete run on a Xeon L5420@2.5GHz).
However, I noticed that ```make``` starts quite a bit of jobs in parallel (```make -j9 [...]```) on top of each build
group being started by GNU ```parallel```. 
I am currently investigating whether this slows down the test process more than it speeds things up.

**Pull request merged**